src/OpenIddict.Client/OpenIddictClientBuilder.cs

```diff
@@ -193,13 +193,24 @@ public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate()
     /// </summary>
     /// <param name="subject">The subject name associated with the certificate.</param>
     /// <returns>The <see cref="OpenIddictClientBuilder"/> instance.</returns>
-    public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500DistinguishedName subject)
+    public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500DistinguishedName subject) =>
+        AddDevelopmentEncryptionCertificate(subject, DateTimeOffset.UtcNow);
+
+    /// <summary>
+    /// Registers (and generates if necessary) a user-specific development encryption certificate.
+    /// </summary>
+    /// <param name="subject">The subject name associated with the certificate.</param>
+    /// <param name="notBefore">The NotBefore of the certificate.</param>
+    /// <returns>The <see cref="OpenIddictClientBuilder"/> instance.</returns>
+    public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500DistinguishedName subject, DateTimeOffset notBefore)
     {
         if (subject is null)
         {
             throw new ArgumentNullException(nameof(subject));
         }
 
+        var notBeforeDateTime = notBefore.DateTime;
```

Comment:
`X509Certificate.NotBefore` and `X509Certificate.NotAfter` always return a "local time" value while `DateTimeOffset.DateTime` is always unspecified, so it's very likely comparisons won't do what we expect. We need a `DateTimeKind.Local` value here (IIRC, `TimeProvider` has a method that returns the "local time").

Reply:
Thank you for clarifying that. I will continue with this in the previous comment:
https://github.com/openiddict/openiddict-core/pull/2071#discussion_r1596864593